app/controllers/phylo_trees_controller.rb

```diff
@@ -278,12 +277,19 @@ def sample_details_json(pipeline_run_ids, taxid)
     # Do not include the query on taxon_counts in the previous query above using a join,
     # because the taxon_counts table is large.
     taxon_counts = TaxonCount.where(pipeline_run_id: pipeline_run_ids).where(tax_id: taxid).index_by { |tc| "#{tc.pipeline_run_id},#{tc.count_type}" }
+
+    metadata_by_sample_id = Metadatum.by_sample_ids(samples_projects.pluck("sample_id"))
+
     samples_projects.each do |sp|
       sp["taxid_reads"] ||= {}
       %w[NT NR].each do |count_type|
         key = "#{sp['pipeline_run_id']},#{count_type}"
         sp["taxid_reads"][count_type] = (taxon_counts[key] || []).count # count is a column of taxon_counts indicating number of reads
       end
+      if metadata_by_sample_id[sp["sample_id"]]
+        sp["sample_type"] = metadata_by_sample_id[sp["sample_id"]][:sample_type]
+        sp["collection_location"] = metadata_by_sample_id[sp["sample_id"]][:collection_location_v2] || metadata_by_sample_id[sp["sample_id"]][:collection_location]
```

Comment:
@jshoe Is this the right way to fetch the location? I noticed that there were old samples that had a value for collection_location but not for collection_location_v2, which is why I did it like this.

Reply:
The `by_sample_ids` should be correct because it'll also grab a location field value. However, you should only have to fetch the `collection_location_v2` field. Do you want to send me the examples that seemed to be missing `collection_location_v2`?